docs/contributors/write-documentation.rst

```diff
@@ -0,0 +1,186 @@
+.. _write documentation:
+
+##########################
+How to write documentation
+##########################
+
+In this document you will learn how to write good, informative, pretty and actionable documentation.
+
+It's not hard!
+
+Overview
+--------
+
+There are two major types of documentation:
+
+1. **docstrings**: your code's docstrings will be automatically parsed by the documentation sofware (`Sphinx <https://www.sphinx-doc.org>`_, more in `about shpinx`_).
+2. **Manual** documentation such as this document.
+
+**Both** are written in `ReStructured Text <https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html>`_ (``.rst``) format.
+
+Some of the great features of using Sphinx is to be able to automatically generate documentation from your code's docstrings, and to be able to link to other parts of the documentation.
+
+For instance: :meth:`~gflownet.gflownet.GFlowNetAgent.trajectorybalance_loss` or to an external function :func:`torch.cuda.synchronize()`.
+
+.. caution::
+
+    This page is about learning the syntax of ``.rst`` and Sphinx. It is not about learning how to write good, informative docstrings.
```

Comment:
Wait...
This says that this page is _not_ about learning how to write good, informative docstrings,
But the comment above (right under the title) says that it _is_ about learning how to write "good, informative [and other stuff] documentation".

Reply:
Yes. I'm making a conceptual difference between 2 things that happen in Sphinx: 

1. rendering regular `.rst` files that explain how to use the package, tutorials, contributors' guides, installation guidelines etc. which I call "documentation".
2. through `autoapi` the automatic exploration, parsing and redering of the package's docstrings

So basically, while `.rst` syntax is used across docstrings AND documentation files, this particular part of the documentation does not explain how to write ddocstrings specifically (for instance: description of what a function does, not HOW it does it ; or using docstring sections with arguments / returns / raises / examples / references)

I'm open to suggestion for a different phrasing :)

Reply:
I've tried to address that concern in [`3ba1cce` (#190)](https://github.com/alexhernandezgarcia/gflownet/pull/190/commits/3ba1ccea38791fa5dfb0dbdc3f416fdc907007f8)

Reply:
Looks good.